src/theme.c

```diff
@@ -189,20 +245,74 @@ load_buttons(struct theme *theme)
 		}
 #endif
 
-		/* If there were no png/svg buttons, use xbm */
+		/* XBM */
 		snprintf(filename, sizeof(filename), "%s.xbm", b->name);
-		alt_filename[0] = '\0';
+		if (!*b->active.buffer) {
+			button_xbm_load(filename, b->active.buffer, b->active.rgba);
+		}
+		if (!*b->inactive.buffer) {
+			button_xbm_load(filename, b->inactive.buffer, b->inactive.rgba);
+		}
+
+		/*
+		 * XBM (alternative name)
+		 * For example max_hover_toggled instead of max_toggled_hover
+		 */
 		if (b->alt_name) {
-			snprintf(alt_filename, sizeof(alt_filename), "%s.xbm", b->alt_name);
+			snprintf(filename, sizeof(filename), "%s.xbm", b->name);
+		}  else {
+			filename[0] = '\0';
+		}
+		if (!*b->active.buffer) {
+			button_xbm_load(filename, b->active.buffer, b->active.rgba);
+		}
+		if (!*b->inactive.buffer) {
+			button_xbm_load(filename, b->inactive.buffer, b->inactive.rgba);
+		}
+
+		/*
+		 * Builtin bitmap
+		 *
+		 * Applicable to basic buttons such as max, max_toggled and
+		 * iconify. There are no bitmap fallbacks for *_hover icons.
+		 */
+		if (!b->fallback_button) {
+			continue;
 		}
 		if (!*b->active.buffer) {
-			button_xbm_load(filename, alt_filename, b->active.buffer,
-				b->fallback_button, b->active.rgba);
+			button_xbm_from_bitmap(b->fallback_button,
+				b->active.buffer, b->active.rgba);
 		}
 		if (!*b->inactive.buffer) {
-			button_xbm_load(filename, alt_filename, b->inactive.buffer,
-				b->fallback_button, b->inactive.rgba);
+			button_xbm_from_bitmap(b->fallback_button,
+				b->inactive.buffer, b->inactive.rgba);
+		}
+	}
+
+	/* Add fallbacks for missing hover overlays */
+	for (size_t i = 0; i < ARRAY_SIZE(buttons); i++) {
+		struct button *button = &buttons[i];
+		if (*button->active.buffer) {
+			continue;
+		}
+
+		/* Given foo_hover, copy foo */
+		char *buf = strdup(button->name);
+		size_t len = strlen(buf);
+		buf[len-6] = '\0';
+		for (size_t j = 0; j < ARRAY_SIZE(buttons); j++) {
+			struct button *tmp = &buttons[j];
+			if (!strcmp(buf, tmp->name)) {
+				create_hover_fallback(theme,
+					button->active.buffer,
+					*tmp->active.buffer);
+				create_hover_fallback(theme,
+					button->inactive.buffer,
+					*tmp->inactive.buffer);
```

Comment:
Same as above, just because the `->active` buffer isn't set it doesn't necessarily mean that the `->inactive` buffer isn't set either for loaders other than the xbm one.